crypto/pqt/pqt_kem.c

```diff
@@ -0,0 +1,498 @@
+                                      uint8_t *shared_secret,
+                                      const uint8_t *public_key,
+                                      const uint8_t *seed) {
+  return (ml_kem_768_ipd_encapsulate_deterministic(ciphertext, shared_secret,
+                                                   public_key, seed) == 0);
+}
+
+static int pq768_decaps(uint8_t *shared_secret, const uint8_t *ciphertext,
+                        const uint8_t *secret_key) {
+  return (ml_kem_768_ipd_decapsulate(shared_secret, ciphertext, secret_key) ==
+          0);
+}
+
+// 1.2 ML-KEM-1024 Wrappers
+// ------------------------
+
+static int pq1024_keygen_deterministic(uint8_t *public_key, uint8_t *secret_key,
+                                       const uint8_t *seed) {
+  return (ml_kem_1024_ipd_keypair_deterministic(public_key, secret_key, seed) ==
+          0);
+}
+
+static int pq1024_encaps_deterministic(uint8_t *ciphertext,
+                                       uint8_t *shared_secret,
+                                       const uint8_t *public_key,
+                                       const uint8_t *seed) {
+  return (ml_kem_1024_ipd_encapsulate_deterministic(ciphertext, shared_secret,
+                                                    public_key, seed) == 0);
+}
+
+static int pq1024_decaps(uint8_t *shared_secret, const uint8_t *ciphertext,
+                         const uint8_t *secret_key) {
+  return (ml_kem_1024_ipd_decapsulate(shared_secret, ciphertext, secret_key) ==
+          0);
+}
+
+// 2. T KEM Wrappers
+// -----------------
+// All wrappers return one on success and zero on error.
+//
+// WARNING: These are internal functions and should not be used outside the PQ/T
+// KEM construction.
+//
+// They do not provide OW-CCA security since the produced shared secret is the
+// raw DH value which does not bind to the ciphertext (the ephemeral public
+// key). And X25519 and NIST-P allow generating different equivalent public
+// keys. So you can recover the shared secret by querying the CCA oracle on a
+// different but equivalent ciphertext.
+//
+// If you need a standalone ECC KEM, consider DHKEM specified in RFC 9180.
+//
+// ASIDE: This insecurity is fine in the PQ/T KEM since they hash the T
+// public key and T ciphertext when deriving the shared secret.
+
+// 2.1 X25518 Wrappers
+// -------------------
+
+static int t25519_keygen_deterministic(uint8_t *public_key, uint8_t *secret_key,
+                                       const uint8_t *seed) {
+  OPENSSL_memcpy(secret_key, seed, PQT_SYMBYTES);
+  X25519_public_from_private(public_key, secret_key);
+  return 1;
+}
+
+static int t25519_encaps_deterministic(uint8_t *ciphertext,
+                                       uint8_t *shared_secret,
+                                       const uint8_t *public_key,
+                                       const uint8_t *seed) {
+  const uint8_t *ephemeral_secret_key = seed;
```

Comment:
`ephemeral_secret_key` and `ephemeral_public_key` are for clarity(?)
Can just rename the parameter in the function signature.